benchmarking_hallucination_metrics/benchmark_hallucination_metrics.ipynb

```diff
@@ -0,0 +1,4732 @@
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/",
+          "height": 304
+        },
+        "id": "o5g_bsRU3J9T",
+        "outputId": "25188e7f-d29f-4429-e3c4-3f20a4f540e2"
+      },
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "source_ds\n",
+              "halueval        10000\n",
+              "DROP             1000\n",
+              "pubmedQA         1000\n",
+              "FinanceBench     1000\n",
+              "covidQA          1000\n",
+              "RAGTruth          900\n",
+              "Name: count, dtype: int64"
+            ]
+          },
+          "execution_count": 8,
+          "metadata": {},
+          "output_type": "execute_result"
+        }
+      ],
+      "source": [
+        "data.source_ds.value_counts()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "IuKKUEGQ3J9T"
+      },
+      "source": [
+        "We won't use all of these datasets in our benchmarks due to quality issues in the underlying labels - read the blog post for more details!\n",
+        "\n",
+        "## Instantiating our hallucination metrics\n",
+        "\n",
+        "We'll instantiate a variety of commonly-used and cutting-edge hallucination metrics to evaluate the performance of our hallucination detection models. \n",
+        "\n",
+        "These metrics include:\n",
+        "\n",
+        "\n",
+        "### Cleanlab's Trustworthy Language Model (TLM)\n",
+        "\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 9,
+      "metadata": {
+        "id": "_8XwGILV3J9U"
+      },
+      "outputs": [],
+      "source": [
+        "import os\n",
+        "from cleanlab_studio import Studio\n",
+        "\n",
+        "studio = Studio(os.environ[\"TLM_API_KEY\"])\n",
+        "tlm = studio.TLM()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Vm94-G4y3J9U"
+      },
+      "source": [
+        "Since `tlm.get_trustworthy_score` takes `prompt` and `response`, we'll turn the `context` and `question` into a single prompt:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 10,
+      "metadata": {
+        "id": "bHfUUFWV3J9U"
+      },
+      "outputs": [],
+      "source": [
+        "data[\"context_and_question_as_prompt\"] = data.apply(\n",
+        "    lambda row: f\"Answer the QUESTION using information **only** from CONTEXT:\\n{row['passage']}\\nQUESTION:\\n{row['question']}\",\n",
+        "    axis=1\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "PAkoqLrO3J9U"
+      },
+      "source": [
+        "## RAGAS instantiation\n",
+        "\n",
+        "We use both the default instantiation of RAGAS, as well as some improvements we've anecdotally found to be useful. We'll call our improved version `RAGAS++`.\n",
```

Comment:
Explicitly state what the improvements in `RAGAS++` are in one place. You mentioned it below but it's hard to grasp since the info is separated in a few diff paragraphs.

Can just combine your text into one paragraph, something like this:
> For our improved `RAGAS++` version, we'll swap out the standard RAGAS LLM (currently `gpt-3.5-turbo` (TODO: this model is from memory, please verify this is actually the default)) with OpenAI's GPT-4o-mini, using the [official RAGAS documentation](https://docs.ragas.io/en/v0.0.22/howtos/customisations/llms.html#evaluating-with-gpt4). We also found that adding a full stop/period character (.), if not already present in the answer, reduces RAGAS runtime errors when run due to the [sentence parsing](https://github.com/explodinggradients/ragas/blob/fe379a1c97d18ce2c203d80432a3da6622337968/src/ragas/metrics/_faithfulness.py#L209-L221) used to create statements. We'll use this for our improved version of RAGAS as well.